src/intelligence_layer/core/model.py

```diff
@@ -406,3 +408,135 @@ def to_instruct_prompt(
         return self.INSTRUCTION_PROMPT_TEMPLATE.to_rich_prompt(
             instruction=instruction, input=input, response_prefix=response_prefix
         )
+
+
+class Message(BaseModel):
+    role: Literal["system", "user", "assistant"]
+    content: str
+
+
+class ChatModel(ABC, AlephAlphaModel):
```

Comment:
I don't think it is trivial to use them in the correct way. Would it make sense to make an initial ChatTask or something similar.

Reply:
I would disagree, it is extremely trivial. It's basically the most common language model API in the world